src/components/auth/torus/AuthTorus.js

```diff
@@ -202,107 +114,55 @@ const AuthTorus = ({ screenProps, navigation, styles, store }) => {
-    })
-    showDialog({
-      onDismiss: hideDialog,
-      content: (
-        <View style={styles.paragraphContainer}>
-          <Paragraph
-            style={[styles.paragraph, styles.paragraphBold]}
-          >{`Hi There,\n did You Mean\n to Signup?`}</Paragraph>
-          <View style={{ marginTop: mainTheme.sizes.defaultDouble }}>
-            <Paragraph
-              style={[styles.paragraph, styles.paragraphContent]}
-            >{`The account doesn’t exist\n or you signed up using`}</Paragraph>
-            <Paragraph style={[styles.paragraphContent, styles.paragraphBold]}>another method</Paragraph>
-          </View>
-        </View>
-      ),
-      buttons: [
-        {
-          text: 'Signup',
-          onPress: () => {
-            fireEvent(SIGNIN_NOTEXISTS_SIGNUP, { provider })
-            hideDialog()
-            resolve('signup')
-          },
-          style: [styles.whiteButton, { flex: 1 }],
-          textStyle: styles.primaryText,
-        },
-        {
-          text: 'Login',
-          onPress: () => {
-            fireEvent(SIGNIN_NOTEXISTS_LOGIN, { provider })
-            hideDialog()
-            resolve('signin')
-          },
-          style: { flex: 1 },
-        },
-      ],
-      buttonsContainerStyle: styles.modalButtonsContainerRow,
-      type: 'error',
-    })
-    return promise
-  }
+  const selfCustodyLogin = useCallback(() => {
+    fireEvent(SIGNIN_METHOD_SELECTED, { method: REGISTRATION_METHOD_SELF_CUSTODY })
+    return navigate('SigninInfo')
+  })
 
   const selfCustody = useCallback(async () => {
     const curSeed = await AsyncStorage.getItem(GD_USER_MASTERSEED)
-    if (isSignup === false) {
-      fireEvent(SIGNIN_METHOD_SELECTED, { method: REGISTRATION_METHOD_SELF_CUSTODY })
-      return navigate('SigninInfo')
-    }
 
     //in case user started torus signup but came back here we need to re-initialize wallet/storage with
     //new credentials
     if (curSeed) {
       await AsyncStorage.clear()
       await ready(true)
     }
+
     fireEvent(SIGNUP_METHOD_SELECTED, { method: REGISTRATION_METHOD_SELF_CUSTODY })
     navigate('Signup', { regMethod: REGISTRATION_METHOD_SELF_CUSTODY })
-  }, [navigate, isSignup])
+  }, [navigate])
 
   const handleLoginMethod = async (
     provider: 'facebook' | 'google' | 'auth0' | 'auth0-pwdless-email' | 'auth0-pwdless-sms',
     torusUserRedirectPromise,
   ) => {
-    showLoadingDialog()
-
-    //in case this is triggered as a callback after redirect we fire a different vent
-    if (torusUserRedirectPromise) {
-      fireEvent(TORUS_REDIRECT_SUCCESS, {
-        isSignup,
-        method: provider,
-        torusPopupMode: torusSDK.popupMode, //this should always be false in case of redirect
-      })
-    } else {
-      fireEvent(isSignup ? SIGNUP_METHOD_SELECTED : SIGNIN_METHOD_SELECTED, {
-        method: provider,
-        torusPopupMode: torusSDK.popupMode, //for a/b testing
-      })
+    if (provider === 'selfCustody') {
```

Comment:
@TomasGonzalez 
self custody is handled this way

handleLoginMathod is called with 'selfCustory' or 'selfCustodyLogin' from the SignupScreen (it have 2 link buttons just after google/fb/pwdless ones)

the functions selfCustory/selfCustodyLogin are performing the corresponding redirect
please keep this logic and try to figure out why this isn't working.

no need to add anythig to the context